CMakeLists.txt

```diff
@@ -1,21 +1,31 @@
 cmake_minimum_required(VERSION 3.2)
-project(fzf C)
+project(fzf LANGUAGES C)
 
 add_library(${PROJECT_NAME} SHARED "src/fzf.c")
 
+set(is-windows $<STREQUAL:${CMAKE_SYSTEM_NAME},Windows>)
+
+target_include_directories(${PROJECT_NAME} PUBLIC
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>)
+target_compile_options(${PROJECT_NAME}
+  PRIVATE
+    $<$<C_COMPILER_ID:MSVC>:/W4>
+    $<$<NOT:$<C_COMPILER_ID:MSVC>>:-Wall>)
+target_compile_definitions(${PROJECT_NAME}
+  PRIVATE
+    $<${is-windows}:_CRT_NONSTDC_NO_DEPRECATE>
+    $<${is-windows}:_CRT_SECURE_NO_DEPRECATE>
+    $<${is-windows}:_CRT_SECURE_NO_WARNINGS>)
+
 set_target_properties(${PROJECT_NAME} PROPERTIES
-    PREFIX "lib"
-    C_STANDARD 99
     WINDOWS_EXPORT_ALL_SYMBOLS ON
-)
-
-target_include_directories(${PROJECT_NAME} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/src)
+    C_STANDARD 99
+    PREFIX lib
+    SUFFIX .so)
```

Comment:
don't we still want a dll on windows?

Reply:
This must have snuck in when I did a rebase. I'll remove the line. 🤦

Reply:
Resolved as of the latest commit.